js/bruh-dash.js

```diff
@@ -127,14 +285,49 @@ global.bruhdash = {
 
   // iterates over elements of a collection and returns all elements that the predicate returns truthy for
   // Note: this should work for arrays and objects
-  filter: function() {
-
+  //returns an array of filtered elements - all the elements that passed the verification condition/function
+
+  filter: function(col, func) {
+    var newEl;
+    var newArr = [];
+    if (Array.isArray(col)){
+      for (var i = 0; i < col.length; i++){
+        if (func(col[i])){
+          newEl = col[i];
+          newArr.push(newEl);
+        }
+      }
+    } else if(typeof col === 'object'){
+        for (var key in col){
+          if (func(col[key]) === true){
+            newEl = col[key];
+            newArr.push(newEl);
+          }
+        }
+    }
+    return newArr;
+    //is this an array of objects or properties of a single object?
+    //what is it returning? - an array? 
+    //the intention of this function when it comes to objects isn't clear to me
   },
 
   // Reduces the collection to a value which is the accumulated result of running each element
   // in the collection through an iteratee
   // Note: this should work for arrays and objects
-  reduce: function() {
-    
+
+  reduce: function(col, func, initial) {
+    var result = 0;
+
+    if (Array.isArray(col)){
+      for (var i = 0; i < col.length; i++){
+        result = func(result,col[i]);
```

Comment:
result should be a summation. Right now you're just replacing what result is at each index of the array. The same is done for objects below.